engine/ddf/src/ddf/ddf_load.cpp

```diff
@@ -178,8 +178,7 @@ namespace dmDDF
 
                     if (field->m_OneOfIndex != DDF_NO_ONE_OF_INDEX)
                     {
-                        FieldDescriptor* field_non_const = (FieldDescriptor*) field;
-                        field_non_const->m_OneOfSet = 1;
+                        message->SetOneOf(desc, field);
```

Comment:
This is basically where the issue is/was. In the old code, we mutated the FieldDescriptor which indeed should be static in memory. I.e, if we have a message containing a oneof block and we use that in more than one serialized message where different oneof values are set, we would effectively mark more than one of these fields as "in use". 

The fix now is that we actually store this information in the messages/structs, so we can keep track of which oneof member is set.